src/client/Microsoft.Identity.Client/TelemetryCore/OpenTelemetry/OtelInstrumentation.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Cache;
+#if SUPPORTS_OTEL
+using System.Diagnostics.Metrics;
+#endif
+using Microsoft.Identity.Client.Internal;
+
+namespace Microsoft.Identity.Client.TelemetryCore.OpenTelemetry
+{
+    /// <summary>
+    /// Class to hold the OpenTelemetry objects used by MSAL.
+    /// </summary>
+    internal class OtelInstrumentation : IOtelInstrumentation
+    {
+        /// <summary>
+        /// Constant to hold the name of the Meter.
+        /// </summary>
+        public const string MeterName = "MicrosoftIdentityClient_Common_Meter";
+
+        /// <summary>
+        /// Constant to holt the name of the ActivitySource.
+        /// </summary>
+        public const string ActivitySourceName = "MicrosoftIdentityClient_Activity";
+
+        private const string SuccessCounterName = "MsalSuccess";
+        private const string FailedCounterName = "MsalFailed";
+        private const string TotalDurationHistogramName = "MsalTotalDuration.1A";
+        private const string TotalDurationInUsHistogramName = "MsalTotalDurationInUs.1A";
+        private const string DurationInCacheHistogramName = "MsalDurationInCache.1A";
+        private const string DurationInHttpHistogramName = "MsalDurationInHttp.1A";
+
+#if SUPPORTS_OTEL
```

Comment:
A different way of achieving this is to create a feature in the Feature directory. Then each TFM that supports OTel inclues this. You probably won't even need "SUPPORTS_OTEL" flag. In code, Only the PlatformProxy will know who supports OTEL and who doesn't.

Reply:
What is feature directory? How can this be achieved? It does make sense to have this only for supported versions.

Reply:
Look at MSAL/platforms/features. A good example is the `DefaultOSBrowser` feature, which is available on desktop versions of .NET only and not on mobile, or `RuntimeBroker` feature, which is available in the package only on net6-win

Reply:
Updated. Thanks!